tests/webserver.py

```diff
@@ -5,26 +5,41 @@
 
 import os
 import threading
-
-try:
-    from http.server import HTTPServer, SimpleHTTPRequestHandler
-except ImportError:
-    from BaseHTTPServer import HTTPServer
-    from SimpleHTTPServer import SimpleHTTPRequestHandler
+from http.server import HTTPServer, SimpleHTTPRequestHandler
 
 
 class MyRequestHandler(SimpleHTTPRequestHandler):
     """Custom HTTP request handler that serves files from another directory."""
 
+    @staticmethod
+    def _chdir(directory):
+        """Context manager to change directory and revert back safely.
+
+        Args:
+        directory (str): The directory to change to.
+
+        Returns:
+            ChangeDir: A context manager that changes the current directory
+            to the specified one and reverts it back upon exit.
+        """
+        class ChangeDir:
+            def __enter__(self):
+                self.old_dir = os.getcwd()
+                os.chdir(os.fspath(directory))
+
+            def __exit__(self, _, __, ___):
+                os.chdir(self.old_dir)
+        return ChangeDir()
+
```

Comment:
This is an interesting approach but adds a lot of complexity that isn't needed. The directory that needs to be served is the same as this file, so there's no need to change directories. 

Also, you should be able to use `pathlib` for the paths.